src/classes/tmDaoTrialFeatureRequest.cls

```diff
@@ -0,0 +1,120 @@
+        return tfr;
+    }
+
+    public static tmTrialFeatureRequest__c initialize(Account acct, Id tenantId, String edition) {
+        //Initialize TFR
+        tmTrialFeatureRequest__c tfr = new tmTrialFeatureRequest__c();
+        tfr.Account__c = acct.Id;
+        tfr.Tenant__c = tenantId;
+        tfr.Edition__c = edition;
+        tfr.Status__c = tmConstants.TFR_STATUS_DRAFT;
+        tfr.StartDate__c = Date.today();
+        if (acct.BillingCountry != null) {
+            tfr.CurrencyIsoCode = tmDaoProvisioning.getCurrencyCode(acct.BillingCountry);
+        }
+        return tfr;
+    }
+
+
+    public static List<tmTrialFeatureRequest__c> getFeatureRequestsByAccount(Id accountId) {
+        return [select tenant__r.CloudServiceProviderId, tenant__r.ExternalId from tmTrialFeatureRequest__c where account__r.Id = :accountId];
+    }
+
+    public static Map<Id, Account> getAccounts(Set<Id> accountIds) {
+        Map<Id, Account> accounts = new Map<Id, Account>([
+            SELECT Id, BillingCity, BillingCountry, BillingPostalCode, BillingState, BillingStreet, RecordType.Name
+            FROM Account
+            WHERE id IN :accountIds]);
+        return accounts;
+    }
+
+    public static String getUserCountry(Id userId) {
+        List<User> u = [SELECT Id, ISO_Country__c FROM user WHERE Isactive = true AND Id = :userId];
+        if (!u.isEmpty() && !String.isEmpty(u[0].ISO_Country__c)) {
+            return u[0].ISO_Country__c;
+        } else {
+            return '';
+        }
+    }
+
+    public static Account updateAccount(Account acct) {
+        tmDaoTrialFeatureRequestHelper tfrHelper = new tmDaoTrialFeatureRequestHelper();
+        return tfrHelper.doUpdateAccountWithoutSharing(acct);
+    }
+
+    public static tmTrialFeatureRequest__c save(tmTrialFeatureRequest__c tfr, List<tmTrialFeatureRequestItem__c> tfrItems) {
+        tmDaoTrialFeatureRequestHelper tfrHelper = new tmDaoTrialFeatureRequestHelper();
+        return tfrHelper.doUpsertTrialFeatureRequestWithoutSharing(tfr, tfrItems);
+    }
+
+    public static void deleteItems(List<tmTrialFeatureRequestItem__c> tfrItems) {
+        tmDaoTrialFeatureRequestHelper tfrHelper = new tmDaoTrialFeatureRequestHelper();
+        tfrHelper.doDeleteTrialFeatureItems(tfrItems);
+    }
+
+
+    private without sharing class tmDaoTrialFeatureRequestHelper {
+        public tmTrialFeatureRequest__c doUpsertTrialFeatureRequestWithoutSharing(tmTrialFeatureRequest__c tfr, List<tmTrialFeatureRequestItem__c> tfrItems) {
+            tfr.AutoApprove__c = isUserProfileInAutoApprovalList();
+            upsert tfr;
+
+            if (tfrItems != null && tfrItems.size() > 0) {
+                delete [SELECT Id FROM tmTrialFeatureRequestItem__c WHERE TrialFeatureRequest__c = :tfr.Id];
+
+                for (tmTrialFeatureRequestItem__c tfrItem : tfrItems) {
+                    tfrItem.TrialFeatureRequest__c = tfr.Id;
+                }
+                insert tfrItems;
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.